custom_model_runner/datarobot_drum/runtime_parameters/runtime_parameters.py

```diff
@@ -109,57 +101,80 @@ class RuntimeParametersLoader:
       awsAccessKeyId: ABDEFGHIJK...
       awsSecretAccessKey: asdjDFSDJafslkjsdDLKGDSDlkjlkj...
       awsSessionToken: null
+    ```
     """
 
-    def __init__(self, values_filepath):
+    ParameterDefinition = namedtuple("ParameterDefinition", ["name", "type", "default"])
+
+    def __init__(self, values_filepath, code_dir):
         if not values_filepath:
             raise InvalidInputFilePath("Empty runtime parameter values file path!")
+        if not code_dir:
+            raise InvalidInputFilePath("Empty code-dir path!")
+
+        self._load_parameter_definitions(code_dir)
```

Comment:
My understanding was that for local development the user does not have to provide a parameter definitions file, to ease his development. However, I do agree that is it much more sustainable and robust to require it.

Reply:
Hmm yeah, that is a good point that it makes things much simpler only have to worry about one file. However, _when I did_ have both files in place when I was doing some local dev, I was tripped up by the fact that my defaults defined in the master file were not being applied as they would when actually deployed on DataRobot.

So I guess I would feel strongly that _if_ the `model-metadata.yaml` file was present, then we should utilize it for local testing; however, I could see an argument for if the file didn't exist, then we can just rely on the values file alone. But, this is a little moot because the user will need to define their params eventually so why not just make them do it up front? I've only made two models so far with runtime params so I don't have a good sense of what the ideal workflow would be yet🤔  I think it is easier to have it required and later relax that restriction than have it relaxed and later have it required and break some peoples workflows so I'm leaning towards this way for now.